src/collection-view.js

```diff
@@ -101,7 +101,7 @@ Marionette.CollectionView = Marionette.View.extend({
   _onCollectionAdd: function(child, collection, opts) {
     var index;
     if (opts.at !== undefined) {
-      index = opts.at;
+      index = opts.index !== undefined ? opts.index : collection.indexOf(child);
```

Comment:
The fact that our unit tests did not catch tells me we need to add a new test :)